packages/libs/sdk/src/api/controllers/nfts.ts

```diff
@@ -108,4 +119,59 @@ export class NftsController {
 
     return formattedResult;
   }
+
+  @QNApolloErrorHandler
+  async getByContractAddress(
+    variables: NFTsByContractAddressQueryVariablesType & NonQueryInput
+  ): Promise<NFTsByContractAddressFormattedResult> {
+    const { chain, ...queryVariables } = variables;
+    const userChain = chain || this.defaultChain;
+    const query: Record<ChainName, TypedDocumentNode<any, any>> = {
+      ethereum: CodegenEthMainnetWalletNFTsByContractAddressDocument,
+      polygon: CodegenPolygonMainnetNFTsByContractAddressDocument,
+      ethereumSepolia: CodegenEthSepoliaWalletNFTsByContractAddressDocument,
+    };
+
+    const {
+      data: {
+        [userChain]: { collection },
+      },
+    } = await this.client.query<
+      NFTsByContractAddressQueryVariablesType, // What the user can pass in
+      NFTsByContractAddressQueryType, // The actual unmodified result from query
+      NFTsByContractAddressQueryResultFull // the modified result (edges and nodes removed)
+    >({
+      query: query[userChain], // The actual graphql query
+      variables: queryVariables,
+      keepTypename: true,
+    });
+
+    if (!collection?.nfts?.length) {
+      return {
+        standard: null,
+        results: [],
+        pageInfo: emptyPageInfo,
+      };
+    }
+
+    const setErcStandard = (
+      results: any
+    ): NFTsByContractAddressFormattedResult => {
+      console.log(results);
+      const standardMap: Record<string, NftErcStandards> = {
+        ERC1155Collection: 'ERC1155',
+        ERC721Collection: 'ERC721',
+      };
+      results['standard'] = standardMap[results['__typename']] || null;
+      delete results['__typename'];
+      return results;
```

Comment:
Can we avoid `delete`? This is just to avoid mutation of the object.
```suggestion
const {__typename, ...newResults} = results;
      return {
         ...newResults,
         standard: standardMap[results['__typename']] || null
         }
```